go/nbs/dynamo_manifest.go

```diff
@@ -39,13 +40,14 @@ type ddbsvc interface {
 type dynamoManifest struct {
 	table, db string
 	ddbsvc    ddbsvc
+	lock      addr
```

Comment:
This is the main thing I hemmed and hawwed about and am still not sure is the right thing. On one hand, the details of this 'lock' hash and how it's calculated and how to check it are totally the concern of the `manifest` implementations. On the other, making it their concern means both `manifest` implementations are now stateful, where before they were stateless, which was nice.